samples/csharp_dotnetcore/05.multi-turn-prompt/DialogExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.BotBuilderSamples
+{
+    public static class DialogExtensions
+    {
+        public static async Task Run(this Dialog dialog, ITurnContext<IMessageActivity> turnContext, ConversationState conversationState, CancellationToken cancellationToken)
+        {
+            var dialogSet = new DialogSet(conversationState.CreateProperty<DialogState>("DialogState"));
+            dialogSet.Add(dialog);
```

Comment:
This is kind of cool, but requires the dialog set to be constructed the same way every time. If the dev for some mysterious reason decides to add logic to choose between two different dialogs here, things will not go well.